content/blog/willing-to-build-up-your-own-workshops-on-demand-infrastructure.md

```diff
@@ -0,0 +1,135 @@
+* D﻿eploy the workshop content on the jupyterhub server in the dedicated student home directory (Notebooks files necessary for the workshop)
+* Send back through API Calls to the frontend server the confirmation of the deployment of the workshop along with the student details (Password)
+
+5﻿.The frontend server get its different tables updated:
+
+* T﻿he customer tables shows an active status for the participant row. The password field has been updated. 
+* T﻿he Workshop table gets also updated. The capacity field decrement the number of available seats. 
+* The student tables gets updated as well by setting the allocated student to active.
+
+6﻿.The frontend server sends the second email to the particpant providing him with the details to connect to the workshop environment.
+
+### T﻿he Run Phase:
+
+F﻿rom the email, the particpant click on the start button. it will open up a browser to the jupyterhub server and directly open up the readme first notebook presenting the workshops flow.
+
+T﻿he participant will go through the different steps and labs of the workshop connecting to the necessary endpoints and leveraging the different kernels available on the jupyterhub server.
+
+M﻿eanwhile, the frontend server will perform regular checks on time passed. Depending on time allocation (from 2 to 4 hours) associated with the workshop, the frontend server will send a reminder email usually a hour before the end of the time allocated. The time count actually starts when the participant hit the register for the workshop button. It is mentionned in the terms and conditions.
+
+F﻿inally, when the time is up: 
+
+T﻿he frontend server sends a new order to the backend to:
+
+P﻿erform either CLEANUP or RESET  action for the dedicated studentid.
+
+T﻿his means:
+
+* Resetting any infrastructure that was requiered for the workshop (Virtual Appliance, Virtual Machine, Docker Container, LDAP config, etc..)
+* G﻿enerate a random Password for the allocated student
+* D﻿elete the workshop content on the jupyterhub server in the dedicated student home directory (Notebooks files necessary for the workshop)
+* Send back through API Calls to the frontend server the confirmation of the CLEANUP or RESET of the workshop along with the student details (Password)
+
+4﻿.The frontend server gets its different tables updated:
+
+* T﻿he customer tables shows an inactive status for the participant row. The password field has been updated. 
+* T﻿he Workshop table gets also updated. The capacity field increment the number of available seats. 
+* The student tables gets updated as well by setting the allocated student to inactive.
+* T﻿he frontend sends the final email to the participant.
+
+### T﻿he React and Reward Phase:
+
+* T﻿he finalemail thanks the customer for its participation.I﻿t provides a link to an external survey link and offers the possibility to share an achievement [badge](https://developer.hpe.com/blog/become-a-legend/) on social media like linkedin or twitter.
```

Comment:
A final email thanks the students for their participation. It provides a link to an external survey and encourages the participants to share their achievement ... on social media, like LinkedIn or Twitter.